### PR TITLE
build: upgrade to .NET 7.0

### PR DESCRIPTION
**Pull request by Flash0ver**
Updating third-party dependencies in #107

### CODE REVIEW

File: source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/DiagnosticAssert.cs [hunk 1]
```diff
@@ -77,9 +78,9 @@ private static void AreEqual(Diagnostic expectedDiagnostic, Diagnostic actualDia
 			errors.AppendLine(nameof(DiagnosticDescriptor.Title), expectedDiagnostic.Descriptor.Title, actualDiagnostic.Descriptor.Title);
 		}
 
-		if (expectedDiagnostic.GetMessage() != actualDiagnostic.GetMessage())
+		if (expectedDiagnostic.GetMessage(CultureInfo.InvariantCulture) != actualDiagnostic.GetMessage(CultureInfo.InvariantCulture))
 		{
-			var message = $"- Message: Expected '{expectedDiagnostic.GetMessage()}', but actually is '{actualDiagnostic.GetMessage()}'.";
+			var message = $"- Message: Expected '{expectedDiagnostic.GetMessage(CultureInfo.InvariantCulture)}', but actually is '{actualDiagnostic.GetMessage(CultureInfo.InvariantCulture)}'.";
```
Comment (Flash0ver) [inline source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/DiagnosticAssert.cs:83]: fixes [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305)

**Comment by NKnusperer** [comment]
Should we bump third-party dependencies as well or will that be done in a separate PR?
```
» F0-Analyzers
  [net7.0]
  Microsoft.CodeAnalysis.Analyzers           3.3.4-beta1.22362.3 -> 3.11.0-beta1.23420.2
  Microsoft.CodeAnalysis.CSharp.Workspaces   4.0.1               -> 4.7.0
  Microsoft.CodeAnalysis.PublicApiAnalyzers  3.3.3               -> 3.3.4
  [.NETStandard,Version=v2.0]
  IndexRange                                 1.0.2               -> 1.0.3
  Microsoft.CodeAnalysis.Analyzers           3.3.4-beta1.22362.3 -> 3.11.0-beta1.23420.2
  Microsoft.CodeAnalysis.CSharp.Workspaces   4.0.1               -> 4.7.0
  Microsoft.CodeAnalysis.PublicApiAnalyzers  3.3.3               -> 3.3.4

» F0.Analyzers
  [.NETFramework,Version=v4.7.2]
  Microsoft.VSSDK.BuildTools  17.0.5234 -> 17.7.2196

» F0.Analyzers.Benchmarks
  [net7.0]
  BenchmarkDotNet                           0.13.2 -> 0.13.8
  Microsoft.CodeAnalysis.CSharp.Workspaces  4.0.1  -> 4.7.0
  [.NETFramework,Version=v4.7.2]
  BenchmarkDotNet                           0.13.2 -> 0.13.8
  Microsoft.CodeAnalysis.CSharp.Workspaces  4.0.1  -> 4.7.0

» F0.Analyzers.Example
  [net7.0]
  Microsoft.Extensions.Hosting  7.0.0 -> 7.0.1

» F0.Analyzers.Example.Tests
  [net7.0]
  coverlet.collector         3.1.2  -> 6.0.0
  Microsoft.NET.Test.Sdk     17.4.0 -> 17.7.2
  MSTest.TestAdapter         2.2.10 -> 3.1.1
  MSTest.TestFramework       2.2.10 -> 3.1.1
  NUnit.Analyzers            3.3.0  -> 3.7.0
  NUnit3TestAdapter          4.2.1  -> 4.5.0
  xunit                      2.4.2  -> 2.5.0
  xunit.runner.visualstudio  2.4.5  -> 2.5.0

» F0.Analyzers.Tests
  [net7.0]
  coverlet.collector                        3.1.2  -> 6.0.0
  FakeItEasy                                7.3.1  -> 7.4.0
  FakeItEasy.Analyzer.CSharp                6.1.0  -> 6.1.1
  FluentAssertions                          6.7.0  -> 6.12.0
  FluentAssertions.Analyzers                0.17.2 -> 0.24.0
  Microsoft.CodeAnalysis.CSharp.Workspaces  4.0.1  -> 4.7.0
  Microsoft.CodeAnalysis.NetAnalyzers       7.0.0  -> 7.0.4
  Microsoft.NET.Test.Sdk                    17.4.0 -> 17.7.2
  MSTest.TestFramework                      2.2.10 -> 3.1.1
  xunit                                     2.4.2  -> 2.5.0
  xunit.runner.visualstudio                 2.4.5  -> 2.5.0
  [.NETFramework,Version=v4.7.2]
  coverlet.collector                        3.1.2  -> 6.0.0
  FakeItEasy                                7.3.1  -> 7.4.0
  FakeItEasy.Analyzer.CSharp                6.1.0  -> 6.1.1
  FluentAssertions                          6.7.0  -> 6.12.0
  FluentAssertions.Analyzers                0.17.2 -> 0.24.0
  Microsoft.CodeAnalysis.CSharp.Workspaces  4.0.1  -> 4.7.0
  Microsoft.CodeAnalysis.NetAnalyzers       7.0.0  -> 7.0.4
  Microsoft.NET.Test.Sdk                    17.4.0 -> 17.7.2
  MSTest.TestFramework                      2.2.10 -> 3.1.1
  xunit                                     2.4.2  -> 2.5.0
  xunit.runner.visualstudio                 2.4.5  -> 2.5.0

» F0.CodeAnalysis.Benchmarking
  [net7.0]
  Microsoft.CodeAnalysis.CSharp.Workspaces  3.3.1 -> 4.7.0
  [.NETStandard,Version=v2.0]
  Microsoft.CodeAnalysis.CSharp.Workspaces  3.3.1 -> 4.7.0

» F0.CodeAnalysis.Testing
  [net7.0]
  Microsoft.CodeAnalysis.CSharp.Workspaces  3.3.1 -> 4.7.0
  [.NETStandard,Version=v2.0]
  Microsoft.CodeAnalysis.CSharp.Workspaces  3.3.1 -> 4.7.0
```

File: source/performance/.globalconfig [hunk 2]
```diff
@@ -2,9 +2,3 @@ is_global = true
 
 # CA1062: Validate arguments of public methods
 dotnet_diagnostic.CA1062.severity = none
-
-# CA1309: Use ordinal string comparison
-dotnet_diagnostic.CA1309.severity = none
```
Comment (Flash0ver) [inline source/performance/.globalconfig:7]: Moved "global suppressions" to local suppressions in source via the [SuppressMessageAttribute](https://learn.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute) for warnings that appear in only a few locations.

File: source/performance/F0.Analyzers.Benchmarks/CodeAnalysis/CodeFixes/DeclareRecordClassExplicitlyBenchmarks.cs [hunk 3]
```diff
@@ -17,11 +17,11 @@ public DeclareRecordClassExplicitlyBenchmarks()
 	[GlobalSetup]
 	public void Setup()
 	{
-		var code =
-@"using System;
+		var code = """
```
Comment (Flash0ver) [inline source/performance/F0.Analyzers.Benchmarks/CodeAnalysis/CodeFixes/DeclareRecordClassExplicitlyBenchmarks.cs:20]: Converted all interpolated and/or verbatim strings that comprise _C# text_ to _raw string literals_.

File: source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/AnalyzerBenchmark.cs [hunk 4]
```diff
@@ -59,7 +60,10 @@ protected static Document CreateDocument(SourceText code, LanguageVersion langua
 
 		var solution = CreateSolution(code, languageVersion, projectId, documentId);
 
-		return solution.GetDocument(documentId);
+		var document = solution.GetDocument(documentId);
+
+		Debug.Assert(document is not null, $"{nameof(DocumentId)} {documentId} is not an ID of a document that is part of this solution.");
```
Comment (Flash0ver) [inline source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/AnalyzerBenchmark.cs:65]: We're getting some new _nullable warnings_ due to multi-targeting _.NET 7.0_ in two projects that previously targeted _.NET Standard 2.0_ only.
```diff
-<TargetFramework>netstandard2.0</TargetFramework>
+<TargetFrameworks>net7.0;netstandard2.0</TargetFrameworks>
```

File: source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/AnalyzerBenchmark.cs [hunk 5]
```diff
@@ -72,12 +76,15 @@ protected static Project CreateProject(SourceText code, LanguageVersion language
...
+		var project = solution.GetProject(projectId);
+
+		Debug.Assert(project is not null, $"{nameof(ProjectId)} {projectId} is not an ID of a project that is part of this solution.");
+		return project;
 	}
 
 	private static Solution CreateSolution(SourceText code, LanguageVersion languageVersion, ProjectId projectId, DocumentId documentId)
 	{
-		var workspace = CreateWorkspace();
+		using var workspace = CreateWorkspace();
```
Comment (Flash0ver) [inline source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/AnalyzerBenchmark.cs:87]: Fixes [CA2000](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2000)

File: source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/CodeFixes/CodeFixBenchmark.cs [hunk 6]
```diff
@@ -77,12 +77,14 @@ public async Task InspectAsync(string expectedCode)
 		var actualText = await formattedDocument.GetTextAsync(CancellationToken.None).ConfigureAwait(false);
 		var actualCode = actualText.ToString();
 
-		if (!actualCode.Equals(expectedText, StringComparison.InvariantCulture))
+		if (!actualCode.Equals(expectedText, StringComparison.Ordinal))
```
Comment (Flash0ver) [inline source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/CodeFixes/CodeFixBenchmark.cs:80]: Fixes [CA1309](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1309).

File: source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/DiagnosticAssert.cs [hunk 7]
```diff
@@ -108,6 +110,7 @@ private static void AreEqual(Diagnostic expectedDiagnostic, Diagnostic actualDia
 		}
 	}
 
+	[SuppressMessage("Globalization", "CA1305:Specify IFormatProvider", Justification = ".NET Standard 2.0")]
```
Comment (Flash0ver) [inline source/performance/F0.CodeAnalysis.Benchmarking/CodeAnalysis/DiagnosticAssert.cs:113]: We're getting some new warnings suggesting to use method overloads from greater TFMs due to multi-targeting _.NET 7.0_ in two projects that previously targeted _.NET Standard 2.0_ only.
```diff
-<TargetFramework>netstandard2.0</TargetFramework>
+<TargetFrameworks>net7.0;netstandard2.0</TargetFrameworks>
```

File: source/production/F0.Analyzers/CodeAnalysis/CodeRefactorings/ObjectInitializer.cs [hunk 8]
```diff
@@ -248,11 +248,11 @@ private static string GetPlainName(IFieldSymbol fieldSymbol)
 	{
 		var name = fieldSymbol.Name;
 
-		if (name.StartsWith("_"))
+		if (name.StartsWith("_", StringComparison.Ordinal))
```
Comment (Flash0ver) [inline source/production/F0.Analyzers/CodeAnalysis/CodeRefactorings/ObjectInitializer.cs:251]: Fixes [CA1310](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1310).

File: source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs [hunk 9]
```diff
@@ -84,92 +86,101 @@ public async Task ComputeRefactoringsAsync_AccessModifiersExternal_AssignsAccess
...
+			{ "Properties with internals", WithFriend(Type_Properties), markup, ObjectInitializer_InternalsAccessible },
+		};
 
 		static string WithFriend(string code)
 		{
-			var Friend =
-				$@"using System.Runtime.CompilerServices;
-					[assembly: InternalsVisibleTo(""{Projects.AssemblyName}"")]";
-			return Friend + Environment.NewLine + Environment.NewLine + code;
+			var Friend = $"""
```
Comment (Flash0ver) [inline source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs:176]: I love _raw string literals_ ❤️

File: source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs [hunk 10]
```diff
@@ -84,92 +86,101 @@ public async Task ComputeRefactoringsAsync_AccessModifiersExternal_AssignsAccess
...
+				using System.Runtime.CompilerServices;
+
+				[assembly: InternalsVisibleTo("{Projects.AssemblyName}")]
+
+				{code}
+				""";
+			return Friend;
```
Comment (Flash0ver) [inline source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs:183]: I love _raw string literals_ ❤️

File: source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs [hunk 11]
```diff
@@ -84,92 +86,101 @@ public async Task ComputeRefactoringsAsync_AccessModifiersExternal_AssignsAccess
...
-				$@"using System.Runtime.CompilerServices;
-					[assembly: InternalsVisibleTo(""{Projects.AssemblyName}"")]";
-			return Friend + Environment.NewLine + Environment.NewLine + code;
+			var Friend = $"""
+				using System.Runtime.CompilerServices;
+
+				[assembly: InternalsVisibleTo("{Projects.AssemblyName}")]
+
+				{code}
+				""";
```
Comment (Flash0ver) [inline source/test/F0.Analyzers.Tests/CodeAnalysis/CodeRefactorings/ObjectInitializerTests.AccessibilityLevels.cs:182]: I love _raw string literals_ ❤️

File: source/test/F0.CodeAnalysis.Testing/CodeAnalysis/CodeFixes/CodeFixTester.cs [hunk 12]
```diff
@@ -3,7 +3,7 @@
 
 namespace F0.Testing.CodeAnalysis.CodeFixes;
 
-internal class CodeFixTester<TDiagnosticAnalyzer, TCodeFix> : CSharpCodeFixTest<TDiagnosticAnalyzer, TCodeFix, XUnitVerifier>
+internal sealed class CodeFixTester<TDiagnosticAnalyzer, TCodeFix> : CSharpCodeFixTest<TDiagnosticAnalyzer, TCodeFix, XUnitVerifier>
```
Comment (Flash0ver) [inline source/test/F0.CodeAnalysis.Testing/CodeAnalysis/CodeFixes/CodeFixTester.cs:6]: Fixes [CA1852](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1852).

**Review by NKnusperer — APPROVED** [review]

**Pull request merged**